embedded/src/main/java/org/wildfly/core/embedded/SystemPropertyContext.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ *
+ * Copyright 2018 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.core.embedded;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+import org.jboss.as.server.ServerEnvironment;
+import org.wildfly.common.Assert;
+import org.wildfly.security.manager.WildFlySecurityManager;
```

Comment:
Almost forgot to address the wildfly-common dependency. That was not already a dependency on the embedded API. I'll remove the references to that so we can make it provided.

Reply:
I ran jdeps on a recent build of the embedded module and it did require wildfly-common even before this change.  It may have previously come in transitively but I think it's a legitimate dependency either way.